laterpay/asset_sources/js/laterpay-backend-shortcode-generator.js

```diff
@@ -113,24 +113,92 @@
 
                     field.state.data.value = value_to_save;
 
-                    var image_element_id = field.settings.preview_element_id || '';
-                    var image_element = $('#' + image_element_id);
+                    var preview_container_id = 'preview_' + field.settings.name;
+                    var preview_container = $( '#' + preview_container_id );
+                    var image_element = $( '.preview-image', preview_container );
+                    var media_name_element = $( '.media-name', preview_container );
+                    var clear_media_element = $( '.button-clear_media', preview_container );
 
                     if ( 0 !== image_element.length ) {
                         var preview_image = attachment.url;
-
-                        if ( 'image' !== attachment.type ) {
-                            preview_image = laterpay_shortcode_generator_labels.no_preview_image;
+                        var media_name = attachment.title;
+
+                        if ( 'video' === attachment.type ) {
+                            preview_image = laterpay_shortcode_generator_labels.preview_images.video;
+                        } else if ( 'audio' === attachment.type ) {
+                            preview_image = laterpay_shortcode_generator_labels.preview_images.audio;
+                        } else if ( 'text' === attachment.type ) {
+                            preview_image = laterpay_shortcode_generator_labels.preview_images.text;
+                        } else if ( 'image' !== attachment.type ) {
+                            preview_image = laterpay_shortcode_generator_labels.preview_images.no_preview_image;
                         }
 
                         $( image_element ).attr( 'src', preview_image );
+                        $( media_name_element ).text( media_name );
+                        $( clear_media_element ).removeClass( 'hidden' );
                     }
 
                 });
 
                 frame.open();
             },
 
+            /**
+             * To generate media preview mark for media field in shortcode generator.
+             *
+             * @param {string} target_field ID of target field.
+             *
+             * @returns {string} HTML markup for media preview.
+             */
+            get_media_markup: function ( target_field ) {
+
+                // @codingStandardsIgnoreStart
+                return '<div id="preview_' + target_field + '" class="lp-media-preview" data-target_field="' + target_field + '">' +
```

Comment:
Does the attribute accept html string only or is jQuery object valid?
If jQuery object is valid maybe create the elements and send it.

Reply:
We have to  return HTML string. I tried create element with JS but that didn't work.